blockchain_storage/src/store.rs

```diff
@@ -265,7 +266,10 @@ impl<'a, Tx: traits::GetMapRef<'a, Schema>> StoreTx<Tx> {
     {
         let col = self.0.get::<DBIdx, I>();
         let data = col.get(key).map_err(crate::Error::from)?;
-        Ok(data.map(|d| T::decode_all(d).expect("Cannot decode a database value")))
+        Ok(data.map(|d| {
+            let mut decode_ptr = d;
+            T::decode_all(&mut decode_ptr).expect("Cannot decode a database value")
+        }))
```

Comment:
Also works:
```rust
Ok(data.map(|mut d| T::decode_all(&mut d).expect("Cannot decode a database value")))
```

Reply:
`mut d` is bad. I found another solution though.